src/ucp/core/ucp_mm.c

```diff
@@ -106,6 +112,36 @@ static ucs_status_t ucp_memh_reg_mds(ucp_context_h context, ucp_mem_h memh,
     return UCS_OK;
 }
 
+ucs_status_t ucp_addr_domain_detect_mds(ucp_context_h context, void *addr, ucp_mem_type_h mem_type_h)
+{
+    ucs_status_t status;
+    unsigned md_index;
+    uct_memory_type_t mem_type = UCT_MD_MEM_TYPE_DEFAULT;
+
+    mem_type_h->md_map = 0;
+    mem_type_h->id = UCT_MD_MEM_TYPE_DEFAULT;
+
+    /*TODO: return if no MDs with address domain detect */
+
+    for (md_index = 0; md_index < context->num_mds; ++md_index) {
+        if (context->tl_mds[md_index].attr.cap.mem_type != UCT_MD_MEM_TYPE_DEFAULT) {
```

Comment:
pls add a shortlist of md's whose memory types have to be checked instead of going over all.
potentially, the implementation of this function+test has to go in a separate PR, and keep this PR for UCT API only.